pdf/pdfcanvas.cpp

```diff
@@ -350,6 +376,57 @@ QPair<QUrl, PDFCanvas::ReducedBox> PDFCanvas::urlAtPoint(const QPointF &point) c
     return QPair<QUrl, PDFCanvas::ReducedBox>();
 }
 
+QPair<Poppler::Annotation*, PDFCanvas::ReducedBox> PDFCanvas::annotationAtPoint(const QPointF &point) const
+{
+    for (int i = 0; i < d->pageCount; ++i) {
+        const PDFPage &page = d->pages.value(i);
```

Comment:
Guess in reality doesn't matter that much, but a bit concerned that now on clicks the document pages are traversed for both urls and links. Can do later, but should consider how to make page lookup and files with ridiculous amount of pages in general more efficient.